src/expansion.rs

```diff
@@ -1,25 +1,32 @@
 use super::peg::Job;
-use super::Variables;
+use super::Shell;
 
-pub fn expand_variables(jobs: &mut [Job], variables: &Variables) {
-    for mut job in &mut jobs[..] {
-        job.command = expand_string(&job.command, variables).to_string();
-        job.args = job.args
-                      .iter()
-                      .map(|original: &String| expand_string(&original, variables).to_string())
-                      .collect();
-    }
+pub trait Expand {
```

Comment:
I appreciate the intent here to turn functions into methods, but I don't think it makes sense in this case. There is no need for the expansion logic to know that `Shell` even exists. There are two things that I think would be reasonable here. 1) Leave it how it was before and have it take a Map of some sort. 2) Create a `Variables` struct. Since option two has other ramifications throughout the code I will add my thoughts as a comment on the PR rather than an inline comment in the file.